File: PAT_Basic/1002 写出这个数.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
	int sum = 0;
	string s;
	cin >> s;
	string piny[] = {"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"};

	for (int i = 0; i < s.length(); i++)
		sum += stoi(s.substr(i, 1));
	string num = to_string(sum);
	
	for (int i = 0; i < num.length(); i++) {
		if(i != 0) cout << " ";
		cout << piny[num[i] - '0'];
	}
	
	return 0;
}

File: PAT_Basic/1002 写出这个数.py
n = input()
cnt = 0
for i in n:
	cnt = cnt + int(i)
temp = ['ling', 'yi', 'er', 'san', 'si', 'wu', 'liu', 'qi', 'ba', 'jiu']
res = str(cnt)
for i in res[:-1]:
	print(temp[int(i)], end = ' ')
print(temp[int(res[-1])])
